packages/node/src/dev-server.mts

```diff
@@ -104,18 +105,20 @@ async function onDevRequest(
     const { headers, body, status } = await handleEvent(req);
     res.statusCode = status;
 
-    for (const [key, value] of headers as unknown as Headers) {
-      // node-fetch does not support headers.getSetCookie(), so we need to
-      // manually set the raw value which can be an array of strings
-      if (value !== undefined) {
-        res.setHeader(key, key === 'set-cookie' ? headers.raw()[key] : value);
-      }
+    for (const [key, value] of headers as Headers) {
+      if (value !== undefined)
+        res.setHeader(
+          key,
+          key === 'set-cookie' ? headers.getSetCookie() : value
+        );
     }
 
-    if (body instanceof Buffer) {
-      res.end(body);
-    } else {
-      body.pipe(res);
+    if (body !== null) {
```

Comment:
If body is null, don't we still need to end the request?

Perhaps something like:

```js
if (body === null) {
  res.end();
}
```